src/components/inputs/toggle/Toggle.tsx

```diff
@@ -0,0 +1,74 @@
+import { motion } from 'framer-motion';
+import { useState } from 'react';
+
+function Toggle({
+  leftLabel,
+  rightLabel,
+  check,
+  className,
+  inputClassName,
+}: {
+  leftLabel: string;
+  rightLabel: string;
+  check: (b: boolean) => void;
+  className?: string;
+  inputClassName?: string;
+}) {
+  const [checked, setChecked] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setChecked(e.target.checked);
+    check(e.target.checked);
+  };
+
+  const longLabel =
+    leftLabel.length > rightLabel.length ? leftLabel : rightLabel;
+
+  return (
+    // eslint-disable-next-line jsx-a11y/label-has-associated-control
+    <label className={`flex items-center ${className}`}>
+      <input
+        id="toggle"
+        type="checkbox"
+        className={`sr-only ${inputClassName}`}
+        onChange={handleChange}
+        checked={checked}
+      />
+      <div className="flex items-center cursor-pointer">
+        <div
+          className={`relative h-8 bg-primary-thin rounded-sm flex items-center justify-between p-1 px-2`}
+          style={{
+            width: longLabel.length * 2 * 11.5,
+          }}
+        >
+          <span
+            className={`z-10 pl-2 transition-all ${
+              !checked ? 'text-black' : 'text-primary'
+            }`}
+          >
+            {leftLabel}
+          </span>
+          <motion.div
```

Comment:
Nice, been meaning to look at framer-motion.

Reply:
recommend checking this one out, its a more serious component I put together as a PoC for the domain cards. Might use it on a grid view for manage assets. https://github.com/ar-io/arns-react/pull/542